### DISCUSSION TITLE
Read 'Inbox' tasks

### DISCUSSION THREAD

**Discussion by Heiinz** [olsh/todoist-net#57] [Q&A] [answered]
I am trying to get my head around how this library interacts with the Todoist API description and would appreciate any guidance to solve this really simple task. 

Here is what I was trying:
string inboxName = "Inbox";
var projects = await _todoistClient.Projects.GetAsync().ConfigureAwait(false);
var projInbox = projects.FirstOrDefault(p => p.Name == inboxName);
if (projInbox is null) throw new ApplicationException($"Cannot find Todoist project '{inboxName}'"); 
var tasks = await _todoistClient.Items.GetAsync(projInbox.Id).ConfigureAwait(false); // <== THROWS EXEPTION

It seems like the last API call expects a 'ComplexId' as parameter where I tried to tell the API call that I want the 'Items' out of my 'Inbox'.

**Comment by olsh** [comment, marked as answer]
`Items.GetAsync` accepts an item ID, not a project ID. Here's how you can get the list of items from the Inbox.

```csharp
var items = await client.Items.GetAsync();
var currentUser = await client.Users.GetCurrentAsync();
var inboxItems = items.Where(i => i.ProjectId == currentUser.InboxProjectId);
```

**Reply by Heiinz** [reply]
@olsh 
Thanks a lot. I would never have found this solution from the available documentation.

**Reply by olsh** [reply]
Unfortunately, the API isn't very intuitive, and I don't have time to document the library. However, your code is almost correct. Here's the fixed version.

```csharp
string inboxName = "Inbox";
var projects = await _todoistClient.Projects.GetAsync().ConfigureAwait(false);
var projInbox = projects.FirstOrDefault(p => p.Name == inboxName);
if (projInbox is null) throw new ApplicationException($"Cannot find Todoist project '{inboxName}'");
var items = await client.Items.GetAsync().ConfigureAwait(false);
var inboxItems = items.Where(i => i.ProjectId == projInbox.Id);
```
